lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -0,0 +1,271 @@
+            event_context = @error_subscriber.events.last.attributes[:context]
+            unmatched = unmatched_attributes(event_context)
+            unless unmatched.empty?
+              return "Expected error attributes to match #{@attributes}, but got these mismatches: #{unmatched} and actual values are #{event_context}"
+            end
+          elsif @error_subscriber.events.empty?
+            return 'Expected the block to report an error, but none was reported.'
+          elsif actual_error.nil?
+            reported_errors = @error_subscriber.events.map { |event| "#{event.error.class}: '#{event.error.message}'" }.join(', ')
+            if @expected_error && @expected_message
+              return "Expected error to be an instance of #{@expected_error} with message '#{@expected_message}', but got: #{reported_errors}"
+            elsif @expected_error
+              return "Expected error to be an instance of #{@expected_error}, but got: #{reported_errors}"
+            elsif @expected_message.is_a?(Regexp)
+              return "Expected error message to match #{@expected_message}, but got: #{reported_errors}"
+            elsif @expected_message.is_a?(String)
+              return "Expected error message to be '#{@expected_message}', but got: #{reported_errors}"
+            end
+          else
+            if @expected_error && !actual_error.is_a?(@expected_error)
+              return "Expected error to be an instance of #{@expected_error}, but got #{actual_error.class} with message: '#{actual_error.message}'"
+            elsif @expected_message
+              return "Expected error message to #{@expected_message.is_a?(Regexp) ? "match" : "be" } #{@expected_message}, but got: '#{actual_error.message}'"
+            else
+              return "Expected specific error, but got #{actual_error.class} with message: '#{actual_error.message}'"
+            end
+          end
+        end
+
+        def failure_message_when_negated
+          error_count = @error_subscriber.events.count
+          error_word = 'error'.pluralize(error_count)
+          verb = error_count == 1 ? 'has' : 'have'
+
+          "Expected the block not to report any errors, but #{error_count} #{error_word} #{verb} been reported."
+        end
+
+        private
+
+        def error_matches_expectation?
+          return true if @expected_error.nil? && @expected_message.nil? && @error_subscriber.events.count.positive?
+
+          @error_subscriber.events.any? do |event|
+            error_class_matches?(event.error) && error_message_matches?(event.error)
+          end
+        end
+
+        def error_class_matches?(error)
+          @expected_error.nil? || error.is_a?(@expected_error)
+        end
+
+        # Check if the given error message matches the expected message pattern
+        def error_message_matches?(error)
+          return true if @expected_message.nil?
+
+          case @expected_message
```

Comment:
There should be existing examples amongst our matchers of something like `value_matches?` that would also allow the use of argument matchers like `a_string_including("network")`.